### ISSUE TITLE
Allow sway-dmenu to consider both name and exec?

### ISSUE THREAD

**Issue by chimak111** [ldelossa/sway-fzfify#7]
For example, 
- the exec `lxappearance` has the name "Customize Look and Feel" but entering `lxa` doesn't pull up the application because neither `a` nor `x` are present in the name.
- the exec `xed` has the name "Text Editor" and that luckily works only because the letters `x`, `e`, and `d` are present.  

Thanks for reading!

**Comment by ldelossa** [comment]
Hey @chimak111 

This script is supposed to work like the "drun" option in Wofi. When using "drun" I don't believe the "exec" field is displayed right ? I think another script would be created if you want to fuzzy match exec commands, but I left this out because I usually just open a terminal to do this.

**Comment by chimak111** [comment]
Hi!

I'm sorry for not being clear! Let me try again.

I'm using Sway on EndeavourOS (an Arch derivative). The config files of many of the Sway/wofi components may be tweaked relative to pure Arch.

Now, with wofi as provided by EndeavourOS,

![wofi](https://user-images.githubusercontent.com/42237881/145708989-20423619-ee95-4368-b9eb-1dcfeb2c2030.gif)

- the input window is empty. "Customize Look and Feel" happens to be fourth from the top
- I type `l` into the input window. "Customize Look and Feel" is now fifth from the top
- I type `x`: only three items remain listed now. "Customize Look and Feel" is at the top
- I type `a`: only "Customize Look and Feel" is listed.

I can do the same with "Archive Manager" (exec =fileroller). By the time I type `filer` only "Archive Manager" remains listed.

So even though only the Name fields are listed by Wofi, the Exec strings can be used to filter even though they are not visible in the list.

**Comment by ldelossa** [comment]
@chimak111 want to give this a shot and see if it has the desired effect? https://github.com/ldelossa/sway-fzfify/tree/dmenu-exec-include


I think if this works, i'd rafactor the script a bit, since the map's value is now in the string itself, probably makes the map not necessary.

**Comment by chimak111** [comment]
Yes!!!

I don't know what you mean by "the map's value is now in the string itself," but if you mean removing the `:` and stuff after that (RHS), while it's true that some names and execs are the same like
- ranger : ranger and
- Galculator : galculator
or like
- Thunar File Manager : thunar

in other cases, the LHS & RHS are not the same but informative. Your call! I'm fine either way.

Thank you once again!

PS: for my personal use and in the context of a desktop launcher, I'd prefer exact matching rather than extended matching.

PPS: with a little fiddling this works just as well on Kubuntu 20.04. Instead of `fd` one has to install `fd-find` and the command is `fdfind`.

**Comment by ldelossa** [comment]
Yup, don't worry about the map comment, I was kinda talking to myself. 

If this works well for you I'll merge it in.

**Comment by ldelossa** [comment]
merged. maybe you want to make a pr which checks for fdfind OR find and uses which ever one it finds?

**Comment by chimak111** [comment]
> ... maybe you want to make a pr which checks for fdfind OR find and uses which ever one it finds?

It's a bit more than just fd/find. Even the last part of the code involving the terminal is alacritty-specific.

So I'm quite happy with sway-dmenu as it is now.

Thanks once again for this cool script!

**Comment by ldelossa** [comment]
Yeah, and I've switched terminals to "kitty" since then. Just a bit lazy/hesitant to map all the sizes cli options like width/height to the various terminals which maybe used. People just seem to be happy with Alacritty.
